golang/x/relay/keeper/keeper_test.go

```diff
@@ -146,11 +143,42 @@ type HeaderTestCases struct {
 	CompareTargets     []CompareCase    `json:"compareTargets"`
 }
 
+/***** Validator TEST CASES *****/
+type ValidateProofTestCase struct {
+	Proof     types.SPVProof      `json:"proof"`
+	BestKnown types.Hash256Digest `json:"bestKnown"`
+	LCA       types.Hash256Digest `json:"lca"`
+	Error     int                 `json:"error"`
+	Output    bool                `json:"output"`
+}
+
+type ValidatorTestCases struct {
+	ValidateProof []ValidateProofTestCase `json:"validateProof"`
+}
+
+/***** Request TEST CASES *****/
+type CheckRequestTestCase struct {
+	InputIdx  uint8          `json:"inputIndex"`
+	OutputIdx uint8          `json:"outputIndex"`
+	Vin       types.HexBytes `json:"vin"`
+	Vout      types.HexBytes `json:"vout"`
+	RequestID types.HexBytes `json:"requestID"`
+	Error     int            `json:"error"`
+	Output    bool           `json:"output"`
+}
+
+type RequestTestCases struct {
+	EmptyRequest  types.ProofRequest     `json:"emptyRequest"`
```

Comment:
do we need this, or can we instantiate a new one on demand?

Reply:
We can't really instantiate an empty one on command because `setRequest` stores the hash of `Pays` and `Spends`.  Meaning if I set an empty Request like this, `s.Keeper.setRequest(s.Context, []byte{0}, []byte{0}, 0, 4)` it gives me `ProofRequest{hashOfSpends, hashOfPays, 0, true, 0}`. I can't simply compare it to `types.ProofRequest{}`.

Reply:
ahhhh gotcha